From 3da8eda0739ae5ec72281fe6fa4eb1605b26fc5b Mon Sep 17 00:00:00 2001
From: Pan Junzhong <junzhong.pan@spacemit.com>
Date: Fri, 22 Dec 2023 14:48:57 +0800
Subject: [PATCH 0229/1206] config: k1-x: enable more usb functions

1. host: enable network adapter, hid, uasp support
2. device: add webcam(f_uvc), uas(f_tcm) support

Change-Id: I4cc10ed14e21e07b7e2b755357502ff76d12da30
---
 arch/riscv/configs/k1-x_evb_defconfig | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/riscv/configs/k1-x_evb_defconfig b/arch/riscv/configs/k1-x_evb_defconfig
index 3ff5ec87c184..685f38af91a0 100644
--- a/arch/riscv/configs/k1-x_evb_defconfig
+++ b/arch/riscv/configs/k1-x_evb_defconfig
@@ -74,6 +74,9 @@ CONFIG_SCSI=y
 CONFIG_BLK_DEV_SD=y
 # CONFIG_BLK_DEV_BSG is not set
 # CONFIG_SCSI_LOWLEVEL is not set
+CONFIG_TARGET_CORE=y
+CONFIG_TCM_IBLOCK=y
+CONFIG_TCM_FILEIO=y
 CONFIG_NETDEVICES=y
 # CONFIG_NET_VENDOR_ALACRITECH is not set
 # CONFIG_NET_VENDOR_AMAZON is not set
@@ -124,6 +127,9 @@ CONFIG_K1X_EMAC=y
 # CONFIG_NET_VENDOR_XILINX is not set
 CONFIG_REALTEK_PHY=y
 CONFIG_CAN_FLEXCAN=y
+CONFIG_USB_USBNET=y
+# CONFIG_USB_NET_AX8817X is not set
+# CONFIG_USB_NET_NET1080 is not set
 # CONFIG_WLAN is not set
 CONFIG_INPUT_EVDEV=y
 # CONFIG_INPUT_KEYBOARD is not set
@@ -307,12 +313,12 @@ CONFIG_SPACEMIT_MIPI_PANEL=y
 CONFIG_POWERVR_ROGUE=y
 CONFIG_FB=y
 # CONFIG_VGA_CONSOLE is not set
-# CONFIG_HID_GENERIC is not set
 CONFIG_USB=y
 CONFIG_USB_XHCI_HCD=y
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_EHCI_K1X=y
 CONFIG_USB_STORAGE=y
+CONFIG_USB_UAS=y
 CONFIG_USB_DWC3=y
 # CONFIG_USB_DWC3_OMAP is not set
 # CONFIG_USB_DWC3_EXYNOS is not set
@@ -330,6 +336,8 @@ CONFIG_USB_CONFIGFS=y
 CONFIG_USB_CONFIGFS_RNDIS=y
 CONFIG_USB_CONFIGFS_MASS_STORAGE=y
 CONFIG_USB_CONFIGFS_F_FS=y
+CONFIG_USB_CONFIGFS_F_UVC=y
+CONFIG_USB_CONFIGFS_F_TCM=y
 CONFIG_MMC=y
 CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_PLTFM=y
-- 
2.47.0

